### PR TITLE
Refactor campaign progression

### PR DESCRIPTION
**Pull request by tkz00**
Waiting on backend PR: https://github.com/lambdaclass/mirra_backend/pull/415

Closes https://github.com/lambdaclass/champions_of_mirra/issues/193

## Motivation

The progress of a player is now a new message from the backend, instead of coming with the user. This PR implements this new communication. Also the campaigns progresses now hold a super campaign id as well as a level and campaign id, this isn't currently used, but will in the future when multiple campaigns are accessible.

## Summary of changes

When getting a user, another call to the backend is made retrieving the campaigns progresses.

## How has this been tested?

Play through levels normally and check the progression works as expected.

### CODE REVIEW

**Review by lotuuu — DISMISSED** [review]

File: client/Assets/Scripts/BackendConnection/SocketConnection.cs [hunk 1]
```diff
@@ -369,6 +374,38 @@ public void CreateUser(string username, Action<User> onGetUserDataReceived)
         SendWebSocketMessage(request);
     }
 
+	public void GetCampaignsProgress(string userId, Action<List<(string, string, string)>> onCampaignProgressReceived)
```
Comment (ncontinanza) [inline client/Assets/Scripts/BackendConnection/SocketConnection.cs:377]: Would the name `GetCampaignProgresses` be more consistent with the rest of the names used here and in the context in which this method is called?

**Review by ncontinanza — CHANGES_REQUESTED** [review]

File: client/Assets/Scripts/BackendConnection/SocketConnection.cs (hunk 1 above)
Comment (tkz00) [reply client/Assets/Scripts/BackendConnection/SocketConnection.cs]: Yeah, it really makes sense. The change is made!

**Review by ncontinanza — DISMISSED** [review]
LGTM!

**Comment by martinacantaro** [comment]
doesn't LGTM :( 

When I go to the Campaign screen for the first time, campaigns 1 and 2 appear as completed even though I never played them. 

Screenshot 2024-04-05 at 12 14 44

**Review by martinacantaro — CHANGES_REQUESTED** [review]
Please fix

**Comment by tkz00** [comment]
> doesn't LGTM :(
> 
> When I go to the Campaign screen for the first time, campaigns 1 and 2 appear as completed even though I never played them.
> 
> One detail to note is that in the several times I shut down the devenv and restarted it, one of those times it magically worked. But 99% of the time it didn't.
> Screenshot 2024-04-05 at 12 14 44

Found the bug and fixed it!

**Review by lotuuu — APPROVED** [review]
Worked great for me! Nice job!

**Review by ncontinanza — APPROVED** [review]

**Review by martinacantaro — APPROVED** [review]
lgtm

**Pull request merged**